lib/business_time/time_extensions.rb

```diff
@@ -196,5 +189,14 @@ def business_time_until(to_time, *currency)
     def during_business_hours?(*currency)
       self.workday?(*currency) && self.to_i.between?(Time.beginning_of_workday(self).to_i, Time.end_of_workday(self).to_i)
     end
+
+    def workday_considering_holidays?(holidays)
+      stringified = strftime("%F")
+      if holidays.all? { |h| h.is_a?(String) }
+        !holidays.include?(stringified)
+      else
+        !holidays.any? { |h| h.to_date == self.to_date }
```

Comment:
1. We could rely on format in which we are storing dates in `BusinessTime::Config.currency_holidays`.
2. Even is we could not, `O(N)` of `holidays.all?`, running _always_, makes all your [premature] optimization senseless:
```ruby
def yyyy_mm_dd
  @yyyy_mm_dd ||= stringified(%F')
end
def workday_considering_holidays?(holidays)
  date = self.to_date
  holidays.none? do |h|
    h.respond_to?(:to_date) ? then h.to_date == date : h.to_s == yyyy_mm_dd
  end
end
```

```